PythonVisualizations/OrderedArray.py

```diff
@@ -252,8 +287,47 @@ def newArraySize(self, val):
         
         self.window.update()
         self.cleanUp(callEnviron)
-        
-    def find(self, val): 
+  #create binary find and linear find
+
+    def find(self,val):
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+        lo = 0  # Point to lo
+        indexLo = self.createIndex(lo, 'lo', level=1)
+        callEnviron |= set(indexLo)
+        hi = len(self.list) - 1  # Point to hi
+        indexHi = self.createIndex(hi, 'hi', level=3)
+        callEnviron |= set(indexHi)
+        mid = (lo + hi) // 2  # Point to the midpoint
+        indexMid = self.createIndex(mid, 'mid', level=2)
+        callEnviron |= set(indexMid)
+        while lo <= hi:
+            mid = (lo + hi) // 2  # Select the midpoint
+            if self.list[mid].val == val:  # Did we find it at midpoint?
+                self.stopAnimations()
+                self.window.update()
+                return mid  # Return the value found
+
+            elif self.list[mid].val < val:  # Is item in upper half?
+                deltaXLo = (mid - lo) + 1
+                self.moveItemsBy(indexLo, (self.CELL_SIZE * deltaXLo, 0))
+                lo = mid + 1  # Yes, raise the lo boundary
+                deltaXMid = ((hi - lo) // 2) + 1
+                self.moveItemsBy(indexMid, (self.CELL_SIZE * deltaXMid, 0))
+
+            else:  # Is item in lower half?
+                deltaXHi = (mid - hi) - 1
+                self.moveItemsBy(indexHi, (self.CELL_SIZE * deltaXHi, 0))
+                hi = mid - 1  # Yes, lower the hi boundary
+                deltaXMid = ((lo - hi) // 2) - 1
+                self.moveItemsBy(indexMid, (self.CELL_SIZE * deltaXMid, 0))
+
+        self.stopAnimations()
+        self.window.update()
+        self.cleanUp(callEnviron)
+        return lo  # val not found
+
+    def binaryFind(self, val):
```

Comment:
The `find()` and `binaryFind()` methods are functionally identical.  They only differ in some whitespace and comments.  There's no need for two functions that do the same thing, and it could cause confusion at some point when a developer makes a change to one and it doesn't affect the other one (which was their intended target).   I think the best way to clean this up is to remove lines 293-329.  That will preserve the `find()` method for the Find button and keep the whitespace in lines 330-366 which lines up the comments on the right.  The call to `binaryFind()` in line 108 should change to `find()`.